app/models/piece.rb

```diff
@@ -18,12 +17,47 @@ def move!(x, y)
     true
   end
 
+  # Return false if coordinates is taken but belongs to friendly piece
+  # otherwise return true
+  def capturable?(x, y)
+    return false unless position_taken?(x, y) && !friendly_piece?(x, y)
+    true
+  end
+
+  # if piece is capturable change piece's attributes
+  def capture!(x, y)
+    return false unless capturable?(x, y)
+    move!(x, y)
+    target_piece(x, y).update_attributes!(
+      x_position: nil,
+      y_position: nil,
+      captured: true
+    )
+  end
```

Comment:
I think we shouldn't combine these 3 methods. Capture and move is a different concept if there is no piece on target coordinates, it would be moved not capture. Also, combine three methods into one method would decrease readability of code. Methods should not be long at all, each method should be transparently obvious and tells a story.